framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -56,99 +53,109 @@ export class CCTransferCommand extends BaseCCCommand {
 
-			return;
+			await this._interopMethod.terminateChain(methodContext, sendingChainID);
 		}
+	}
+
+	public async execute(ctx: CCCommandExecuteContext): Promise<void> {
+		const { ccm, ccmSize } = ctx;
+		const methodContext = ctx.getMethodContext();
+		const { id: ownChainID } = await this._interopMethod.getOwnChainAccount(methodContext);
+		let params: CCTransferMessageParams;
+		const { sendingChainID, status, fee } = ccm;
+		let recipientAddress: Buffer;
+
+		try {
+			params = codec.decode<CCTransferMessageParams>(crossChainTransferMessageParams, ccm.params);
+			validator.validate(crossChainTransferMessageParams, params);
+		} catch (error) {
+			ctx.logger.debug({ err: error as Error }, 'Error verifying the params.');
+
+			if (status === CCM_STATUS_OK && fee >= MIN_RETURN_FEE * ccmSize) {
+				await this._interopMethod.error(methodContext, ccm, CCM_STATUS_PROTOCOL_VIOLATION);
+			}
 
-		if (
-			!tokenSupported(this._supportedTokenIDs, params.tokenID) &&
-			ccm.fee >= ctx.ccmSize * MIN_RETURN_FEE &&
-			ccm.status === CCM_STATUS_OK
-		) {
-			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_TOKEN_NOT_SUPPORTED);
+			await this._interopMethod.terminateChain(methodContext, sendingChainID);
 			return;
 		}
+		const { tokenID, amount, senderAddress } = params;
+		recipientAddress = params.recipientAddress;
+		const [tokenChainID, tokenLocalID] = splitTokenID(tokenID);
 
-		let { recipientAddress } = params;
-		if (ccm.status !== CCM_STATUS_OK) {
-			recipientAddress = params.senderAddress;
+		if (!tokenSupported(this._supportedTokenIDs, tokenID)) {
+			// TODO: emit event else throw error
 		}
 
-		const canonicalTokenID = await this._tokenMethod.getCanonicalTokenID(
-			methodContext,
-			params.tokenID,
-		);
-		const recipientExist = await this._tokenMethod.accountExists(
-			methodContext,
-			params.recipientAddress,
-		);
-		let receivedAmount = params.amount;
-		if (!recipientExist) {
-			const minBalance = this._minBalances.find(mb => mb.tokenID.equals(canonicalTokenID))?.amount;
-			if (!minBalance || minBalance > params.amount) {
-				if (ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize && ccm.status === CCM_STATUS_OK) {
-					await this._interopMethod.error(methodContext, ccm, CCM_STATUS_MIN_BALANCE_NOT_REACHED);
-				}
-				return;
-			}
-			receivedAmount -= minBalance;
-			const [canonicalChainID, canonicalLocalID] = splitTokenID(canonicalTokenID);
-			if (canonicalChainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
-				const supplyStore = this.stores.get(SupplyStore);
-				const supply = await supplyStore.get(methodContext, canonicalLocalID);
-				supply.totalSupply -= minBalance;
-				await supplyStore.set(methodContext, canonicalLocalID, supply);
+		if (status !== CCM_STATUS_OK) {
+			recipientAddress = senderAddress;
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const user = await userStore.get(methodContext, userStore.getKey(recipientAddress, tokenID));
```

Comment:
`get` will throw with `NotFoundError`, so if you want to handle non existing case, you need to handle like below
```
try {
  const user = await userStore.get(methodContext, userStore.getKey(recipientAddress, tokenID));
} catch (error) {
  if (!(error instanceof NotFoundError)) {
    throw error
  }
  // do not found error case
}
```
but maybe
```
const recipientExist = await userStore.has(methodContext, userStore.getKey(recipientAddress, tokenID));
```
is sufficient in this case